VDA-DID-Registry/contracts/VeridaDIDRegistry.sol

```diff
@@ -5,419 +5,80 @@ pragma solidity ^0.8.6;
-        unsignedMsg = abi.encodePacked(
-          unsignedMsg,
-          attributeParams[i].name,
-          attributeParams[i].value,
-          attributeParams[i].validity,
-          attributeParams[i].proof
-        );
-      }
-      unsignedMsg = abi.encodePacked(unsignedMsg, nonce[identity]);
-      require(VeridaDataVerificationLib.validateSignature(unsignedMsg, signature, identityOwner(identity)), "Invalid Signature");
-      nonce[identity]++;
-    }
-    _bulkAdd(identity, delegateParams, attributeParams);
-  }
+    uint length = list.length();
+    string[] memory ret = new string[](length);
 
-  /**
-   * @notice Perform 'revoke' operations of multiple delegates & attributes
-   * @dev Only called after checking the transaction signature
-   * @param identity - DID
-   * @param delegateParams - array of delegates to be added
-   * @param attributeParams - array of attributes to be added
-   */
-  function _bulkRevoke(
-    address identity,
-    RevokeDelegateParam[] calldata delegateParams,
-    RevokeAttributeParam[] calldata attributeParams
-  ) internal {
-    for (uint i = 0; i < delegateParams.length; i++) {
-      _revokeDelegate(identity, delegateParams[i].delegateType, delegateParams[i].delegate);
+    for (uint i = 0; i < length; i++) {
+      ret[i] = list.at(i);
     }
 
-    for (uint i = 0; i < attributeParams.length; i++) {
-      _revokeAttribute(identity, attributeParams[i].name, attributeParams[i].value);
-    }
+    return ret;
   }
 
   /**
-   * @notice Perform 'revoke' operations of multiple delegates & attributes
-   * @dev Check transaction signature and call internal function
-   * @param identity - DID
-   * @param delegateParams - array of delegates to be added
-   * @param attributeParams - array of attributes to be added
-   * @param signature - transaction signature
+   * @dev See {IVeridaDIDRegistry}
    */
-  function bulkRevoke(
-    address identity,
-    RevokeDelegateParam[] calldata delegateParams,
-    RevokeAttributeParam[] calldata attributeParams,
-    bytes calldata signature
-  ) external {
-    {
-      bytes memory unsignedMsg = abi.encodePacked(identity);
-      for (uint i = 0; i < delegateParams.length; i++) {
-        unsignedMsg = abi.encodePacked(
-          unsignedMsg,
-          delegateParams[i].delegateType,
-          delegateParams[i].delegate
-        );
-      }
-
-      for (uint i = 0; i < attributeParams.length; i++) {
-        unsignedMsg = abi.encodePacked(
-          unsignedMsg,
-          attributeParams[i].name,
-          attributeParams[i].value
-        );
-      }
-      unsignedMsg = abi.encodePacked(unsignedMsg, nonce[identity]);
-      require(VeridaDataVerificationLib.validateSignature(unsignedMsg, signature, identityOwner(identity)), "Invalid Signature");
-      nonce[identity]++;
-    }
-    _bulkRevoke(identity, delegateParams, attributeParams);
+  function nonce(address didAddress) external view override returns(uint) {
```

Comment:
Need to throw an error if the didAddress not found.

Reply:
Resolved.